### ISSUE TITLE
Restore security functionality to the SDK

### ISSUE THREAD

**Issue by ryanbogan** [opensearch-project/opensearch-sdk-java#298]
All aspects from feature/identity  (specifically tokens) that were in the SDK have now been temporarily removed, since the extensibility team is working off the main branch.  Once the feature/identity branch is merged into main, this functionality must be added back in.

**Comment by saratvemulapalli** [comment]
@peternied @cwperks extensions has moved all code to OpenSearch main, so we would need to get feature/identity to main to consume the changes for all things security.
